<a href="https://colab.research.google.com/github/ChiruChirag/CUSTOMER_CHURN_ANALYSIS/blob/main/EmployeeProject.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

**CHURN ANALYLIS OF EMPLOYEES OF A PARTICULAR COMPANY**

In [3]:
import pandas as pd
import pylab as pl
import numpy as np
import scipy.optimize as opt
from sklearn import preprocessing
%matplotlib inline 
import matplotlib.pyplot as plt
df=pd.read_csv("https://cf-courses-data.s3.us.cloud-object-storage.appdomain.cloud/IBMDeveloperSkillsNetwork-ML0101EN-SkillsNetwork/labs/Module%203/data/ChurnData.csv")
#display(df)   OPTIONAL U CAN DISPLAY IT TO SEE THE DATA SET
dff = df[["ed","loglong","equip","longmon","callcard"]]
dff.head()
X = np.asanyarray(dff[["ed","loglong","equip","longmon","callcard"]])
X[0:5]
y = np.asanyarray(df['churn'])
y [0:5]
from sklearn import preprocessing
X = preprocessing.StandardScaler().fit(X).transform(X)
X[0:5]
from sklearn.model_selection import train_test_split#TEST AND TRAINING SET CREATION
X_train, X_test, y_train, y_test = train_test_split( X, y, test_size=0.2, random_state=4)
print ('Train set:', X_train.shape,  y_train.shape)
print ('Test set:', X_test.shape,  y_test.shape)
from sklearn.linear_model import LogisticRegression
from sklearn.metrics import confusion_matrix
LR = LogisticRegression(C=0.01, solver='liblinear').fit(X_train,y_train)
LR
yhat = LR.predict(X_test)
yhat
yhat_prob = LR.predict_proba(X_test)
yhat_prob
array=confusion_matrix(y_test, yhat)
a=(array[0,0]+array[1,1])/((array[0,0]+array[0,1]+array[1,0]+array[1,1]))#THIS IS TO FIND THE ACCURACY
display(a);


Train set: (160, 5) (160,)
Test set: (40, 5) (40,)


0.775

I HAVE GOT AN ACURACY OF 0.775 WHICH IS BARICALLY 77.5% USING THIS PARTICULAR SET OF DATA
WE CAN TRY VARIOUS DATA'S AND TRY TO INCREASE THE PROBILITY OF OUR PREDICTION